app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -391,7 +391,10 @@ public void onItemClick(AdapterView<?> parent, View view,
                 if (id == allPeopleId) {
                     doNarrow(new NarrowFilterAllPMs(app.getYou()));
                 } else {
-                    narrow_pm_with(Person.getById(app, (int) id));
+                    Person person = Person.getById(app, (int) id);
+                    narrow_pm_with(person);
+                    switchToPrivate();
+                    setRecipientsInChatBox(person.getEmail());
```

Comment:
Nice, the code could be reduced though. ZulipActivity implements [NarrowListener](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowListener.java) which has a function onNarrowFillSendBoxPrivate to perform this job :)

Also on narrowing to stream, the chatbox is not populated when I click on views except [stream name textview](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java#L1152)